tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php

```diff
@@ -0,0 +1,196 @@
+            'country_code'    => 'US',
+            'phone'           => '+1 123 111 3333',
+            'phone_number'    => '+1 123 111 3333',
+        ];
+
+        $postData = [
+            'shipping_address' => [
+                'street' => [
+                    'Test street 123',
+                    'Additional street 007',
+                ],
+                'firstname'     => 'TesterFn',
+                'lastname'      => 'TesterLn',
+                'city'          => 'Los Angeles',
+                'region_id'     => 12, // Region: CA
+                'postcode'      => '10011',
+                'country_id'    => 'US',
+                'telephone'     => '+1 123 111 3333',
+            ]
+        ];
+
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $result = $this->currentMock->prepareAddressData($postData);
+
+        $this->assertEquals($actual, $result);
+    }
+
+    public function testSaveActionWhereBoltReferenceIsFalse()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest', 'getResponse', '_redirect'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', false);
+
+        $response = new Mage_Core_Controller_Response_Http();
+
+        $this->currentMock->method('getRequest')
+            ->willReturn($request);
+
+        $this->currentMock->method('getResponse')
+            ->willReturn($response);
+
+        $this->currentMock->method('_redirect')
+            ->willReturn(false);
+
+        $result = $this->currentMock->saveAction();
+
+        $this->assertNull($result);
+    }
+
+    public function testSaveAction()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods([
+                'getRequest', '_redirect', '_processActionData', '_getSession',
+                'getImmutableQuoteIdFromTransaction', '_normalizeOrderData',
+                '_getOrderCreateModel', '_postOrderCreateProcessing'
+            ])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $session = new Mage_Adminhtml_Model_Session_Quote();
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', true);
+        $request->setPost('payment', false);
+
+        $this->currentMock->method('getRequest')
+            ->willReturn($request);
+
+        $this->currentMock->method('_normalizeOrderData')
+            ->willReturn(false);
+        $this->currentMock->method('_processActionData')
+            ->willReturn(true);
+
+        $this->currentMock->method('_redirect')
```

Comment:
can we also check redirect url?

Reply:
I have no idea how to check redirect. Can you give an advice?

Reply:
can you capture the argument of method call with phpunit?